### PR TITLE
[OSSM-3621] Add the ability to fetch released tags instead of the HEAD

### PR DESCRIPTION
**Pull request by leoieggli**

### CODE REVIEW

File: tools/automator-main.sh [hunk 1]
```diff
@@ -226,6 +230,10 @@ validate_opts() {
   if [ -z "${git_exclude:-}" ]; then
     git_exclude=""
   fi
+
+  if [ -n "${fetch_tag:-}" ]; then
+    fetch_tag=true
+  fi
```
Comment (jwendell) [inline tools/automator-main.sh:236]: What did you want to achieve here?

File: tools/automator-main.sh [hunk 2]
```diff
@@ -279,6 +287,11 @@ commit() {
 }
 
 merge() {
+  # If fetch_tags is TRUE, then merge_branch var must use the latest tag from the branch
+  if [ -n "${fetch_tags:-}" ]; then
+    merge_branch=$(git describe --tags --abbrev=0 "$branch")
```
Comment (jwendell) [inline tools/automator-main.sh:292]: You probably wanted to use the `$merge_branch` variable instead of `$branch`.

**Review by jwendell — CHANGES_REQUESTED** [review]
You are mixing `fetch_tag` with `fetch_tags`. This is a clear indicative this was not tested.

Let's schedule a sync call so we can test it together.

File: tools/automator-main.sh (hunk 1 above)
Comment (leoieggli) [reply tools/automator-main.sh:241]: Removed.

File: tools/automator-main.sh [hunk 3]
```diff
@@ -16,6 +16,7 @@
 # limitations under the License.
 
 set -euo pipefail
+# set -x
```
Comment (jwendell) [inline tools/automator-main.sh:19]: remove this

File: tools/automator-main.sh [hunk 4]
```diff
@@ -226,6 +231,15 @@ validate_opts() {
   if [ -z "${git_exclude:-}" ]; then
     git_exclude=""
   fi
+
+  if $fetch_tags; then
```
Comment (jwendell) [inline tools/automator-main.sh:235]: this will fail if `--fetch-tags` is not passed in the command line. I suggest to initialize `fetch_tags=false` at the top the `validate_opts` function

File: tools/automator-main.sh [hunk 5]
```diff
@@ -226,6 +231,15 @@ validate_opts() {
...
+    if [ -n "$merge_branch" ]; then
+      tag_branch="$(git describe --tags --abbrev=0 "$merge_branch")"
```
Comment (jwendell) [inline tools/automator-main.sh:237]: Is this happening at the right place? At this point we didn't even clone the repo, I think the `git describe` is running against this (test-infra) repo, not the target one.

**Review by jwendell — CHANGES_REQUESTED** [review]

File: tools/automator-main.sh (hunk 3 above)
Comment (leoieggli) [reply tools/automator-main.sh]: Done.

File: tools/automator-main.sh (hunk 4 above)
Comment (leoieggli) [reply tools/automator-main.sh]: Adjusted.

File: tools/automator-main.sh (hunk 5 above)
Comment (leoieggli) [reply tools/automator-main.sh]: I was doing the login in another part. I removed from here and add it into the merge function. The test passed using the parameter and without it (using my local repos), but please validate it too if possible.

File: tools/automator-main.sh [hunk 6]
```diff
@@ -226,6 +234,7 @@ validate_opts() {
   if [ -z "${git_exclude:-}" ]; then
     git_exclude=""
   fi
+
```
Comment (jwendell) [inline tools/automator-main.sh:237]: Why did you remove the check when `fetch_tags` is present, then `merge` must be `true`?

File: tools/automator-main.sh [hunk 7]
```diff
@@ -281,7 +290,14 @@ commit() {
 merge() {
   local src_branch="${AUTOMATOR_SRC_BRANCH:-none}"
   fork_name="$src_branch-$branch-$modifier-$(hash "$title")"
-  git remote add -f -t "$merge_branch" upstream "$merge_repository"
+
+  if $fetch_tags; then
+    merge_branch="$(git describe --tags --abbrev=0 "$merge_branch")"
```
Comment (jwendell) [inline tools/automator-main.sh:295]: this `if` statement could end here.

That's all you need. The `git remote add` command from above goes below.

In code:
```sh
if $fetch_tags; then
  merge_branch="$(git describe --tags --abbrev=0 "$merge_branch")"
fi
git remote add -f -t "$merge_branch" upstream "$merge_repository"
```

```

**Review by jwendell — CHANGES_REQUESTED** [review]

File: tools/automator-main.sh (hunk 6 above)
Comment (leoieggli) [reply tools/automator-main.sh:237]: Restored.

File: tools/automator-main.sh (hunk 7 above)
Comment (leoieggli) [reply tools/automator-main.sh]: Done. But it is failing on my machine (maybe because of my forks):

```
+ true
+ '[' -n maistra-2.3 ']'
++ git describe --tags --abbrev=0 maistra-2.3
+ merge_branch=maistra-2.3.1
+ git remote add -f -t maistra-2.3.1 upstream https://github.com/leoieggli/api
Updating upstream
fatal: couldn't find remote ref refs/heads/maistra-2.3.1
error: Could not fetch upstream
```

Comment (jwendell) [reply tools/automator-main.sh]: If it's failing the it's not correct. Please schedule a call so we can test this together.

Comment (leoieggli) [reply tools/automator-main.sh]: Fixed. The script worked - the git remote add is a little bit different  when using tags. So, the code will be like this:

`  if $fetch_tags; then
    if [ "${merge_strategy}" == "merge" ]; then
      merge_branch="$(git describe --tags --abbrev=0 "$merge_branch")"
      git remote add upstream $merge_repository
      git fetch upstream tag $merge_branch
      echo "Using tag $merge_branch"
    else
      print_error_and_exit "merge_strategy value must be 'merge', and is required when using '--fetch-tags' option"
    fi
  else
    git remote add -f -t "$merge_branch" upstream "$merge_repository"
    echo "Using branch $merge_branch"
  fi`

**Review by jwendell — APPROVED** [review]
